src/gui/src/qpxgraph.cpp

```diff
@@ -0,0 +1,1403 @@
+
+  if (eflags) {
+    if (ttype == TEST_TA) {
+      taL = taLayer;
+      taZ = taZone;
+      taLayer = (eflags & (0xF << 28)) >> 28;
+      taZone = (eflags & (0x0FFFFFF0)) >> 4;
+    } else {
+      errcList = eflags;
+      showspeed = 0;
+    }
+  }
+#if 0
+#ifndef QT_NO_DEBUG
+    qDebug(QString("drawGraph: rect: %1.%2 - %3.%4")
+           .arg(rect.left())
+           .arg(rect.top())
+           .arg(rect.right())
+           .arg(rect.bottom())
+          );
+    qDebug(QString("drawGraph: grect: %1.%2 - %3.%4")
+           .arg(grect.left())
+           .arg(grect.top())
+           .arg(grect.right())
+           .arg(grect.bottom())
+          );
+#endif
+#endif
+
+  p->save();
+  if (ttype == TEST_TA) {
+    sg = QSize(s.width() - 2, s.height() - margin_bottom - 1);
+    if (forceAll)
+      p->fillRect(1, 1, width() - 2, height() - margin_bottom - 2,
+                  QBrush(settings->col_bg));
+  } else {
+    sg = QSize(s.width() - (margin_left + margin_right + 2),
+               s.height() - margin_bottom - 1);
+    if (forceAll || margin_left > rect.x())
+      p->fillRect(margin_left + 1, 1,
+                  width() - (margin_left + margin_right + 2),
+                  height() - margin_bottom - 1, QBrush(settings->col_bg));
+    else
+      p->fillRect(rect.x(), 1, width() - rect.x() - margin_left,
+                  height() - margin_bottom - 1, QBrush(settings->col_bg));
+
+    Vscale = (s.height() - margin_bottom) / 60.0;
+    if (dev->media.type.startsWith("CD")) {
+      HscaleLBA = 450000 / sg.width();
+      Vscale1X = Vscale;
+    } else if (dev->media.type.startsWith("DVD")) {
+      HscaleLBA = (1 << 19) * 5 * dev->media.ilayers / sg.width();
+      Vscale1X = Vscale * 3;
+    } else if (dev->media.type.startsWith("BD")) {
+      HscaleLBA =
+          (1 << 19) * dev->media.igbpl * dev->media.ilayers / sg.width();
+      Vscale1X = Vscale * 4;
+    }
+
+    p->translate(margin_left + 1, 1);
+  }
+
+  switch (ttype) {
+  case TEST_RT:
+    drawTransfer(p, sg, dev, grect, 0);
+    break;
+  case TEST_WT:
+    drawTransfer(p, sg, dev, grect, 1);
+    break;
+  case TEST_ERRC:
+    drawErrc(p, sg, dev, grect);
+    break;
+  case TEST_JB:
+    drawJB(p, sg, dev, grect);
+    break;
+  case TEST_FT:
+    drawFT(p, sg, dev, grect);
+    break;
+  case TEST_TA:
+    drawTA(p, sg, dev, grect);
+    break;
+  default:
+    break;
+  }
+  p->restore();
+  if (ttype == TEST_TA) {
+    drawGridTA(p, s, dev, ttype);
+  } else {
+    drawGrid(p, s, dev, ttype);
+  }
+  if (eflags) {
+    if (ttype == TEST_TA) {
+      taLayer = taL;
+      taZone = taZ;
+    } else {
+      errcList = el;
+      showspeed = ss;
+    }
+  }
+}
+
+void QPxGraph::drawTransfer(QPainter *p, const QSize &s, device *dev,
+                            const QRect &rect, bool rw) {
+  Q_UNUSED(rect);
+
+  IntList xerr;
+  QPainterPath pp;
+  int x, xo = 0;
+  int startidx = 0;
+  int i;
+  uint64_t lastLBA = 0;
```

Comment:
Codacy found an issue: [Variable 'lastLBA' is assigned a value that is never used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)